### PR TITLE
Tensorboard

### PR DESCRIPTION
**Pull request by krishanr**
This pull request adds tensorboard logging, which includes logging of various metrics, hyperparameters and the model architecture.

### CODE REVIEW

File: train_segmentation.py [hunk 1]
```diff
@@ -189,252 +231,352 @@ def main(params, config_path):
...
 
     now = datetime.datetime.now().strftime("%Y-%m-%d_%H-%M")
-    num_classes = params['global']['num_classes']
-    task = params['global']['task']
-    assert task == 'segmentation', f"The task should be segmentation. The provided value is {task}"
-    num_classes_corrected = num_classes + 1  # + 1 for background # FIXME temporary patch for num_classes problem.
+    num_classes = params["global"]["num_classes"]
+    task = params["global"]["task"]
+    assert task == "segmentation", f"The task should be segmentation. The provided value is {task}"
+    num_classes_corrected = (
```
Comment (CharlesAuthier) [inline train_segmentation.py:244]: Why did you change that line?

**Review by CharlesAuthier — COMMENTED** [review]
Be sure that the code pass the Travis tests please, other than that good job.

**Review by victorlazio109 — COMMENTED** [review]
Similar logging functionality offered by tensorboard is implemented with mlflow which tracks the life-cyle of ML experiments in entirety. This duplicates already existing logging functionality.
